jetstream_pt/layers.py

```diff
@@ -115,23 +124,37 @@ def quantize_weight_from_nn_linear(self, weight):
         self.in_features,
     ), f"Got unexpected weight of shape {weight.shape}, expected weight shape ({self.out_features}, {self.in_features})."
     w_q, scale, zp = quantize_tensor(
-        weight, (1,), self.n_bit, self.is_symmetric, block_size=-1
+        weight, (1,), self.n_bit, self.is_symmetric_weight, block_size=-1
     )
     w_dq = dequantize_tensor(w_q, scale, zp)
     self._load_quantized_weights(w_q, scale, zp)
 
   def forward(self, inputs):
     if not self.run_fake_quantize:
-      if self.is_symmetric:
-        return torch.mul(F.linear(inputs, self.weight), self.weight_scaler)
+      if self.quantize_activation:
+        inputs, act_s, _ = quantize_tensor(inputs, reduce_axis=(2,))
+      if not self.quantize_activation:
+        result = F.linear(inputs, self.weight)
       else:
-        out = torch.mul(F.linear(inputs, self.weight), self.weight_scaler)
+        result = torchjax.call_jax(
```

Comment:
Here we have to call jax because we need to do `dot(int8, int8)->int32`. This semantic cannot be represented in torch now. In torch, the inferred output dtype of 2 `int8` operands will be `int8`, causing overflow on the dot result. The `dot_general` in JAX support specifying output dtype, hence we use it here.

Let me add a comment here.